controllers/integration/integration_adapter.go

```diff
@@ -137,6 +137,143 @@ func (a *Adapter) getAllApplicationSnapshots() (*[]appstudioshared.ApplicationSn
 	return &applicationSnapshots.Items, nil
 }
 
+func (a *Adapter) getAllEnvironments() (*[]appstudioshared.Environment, error) {
+
+	environmentList := &appstudioshared.EnvironmentList{}
+	opts := []client.ListOption{
+		client.InNamespace(a.application.Namespace),
+	}
+	err := a.client.List(a.context, environmentList, opts...)
+	return &environmentList.Items, err
+}
+
+func (a *Adapter) findOneAvailabeEnvironment() (*appstudioshared.Environment, error) {
+	allEnvironments, err := a.getAllEnvironments()
+	if err != nil {
+		return nil, err
+	}
+	for _, environment := range *allEnvironments {
+		if environment.Spec.ParentEnvironment == "" {
+			found := 1
+			for _, tag := range environment.Spec.Tags {
+				if tag == "ephemeral" {
+					found = 0
+					break
+				}
+			}
+			if found == 1 {
+				return &environment, nil
+			}
+		}
+	}
+	return nil, nil
+}
+
+func (a *Adapter) findMatchingComponents() (*[]hasv1alpha1.Component, error) {
```

Comment:
We already have the [getAllApplicationComponents](https://github.com/redhat-appstudio/integration-service/blob/main/controllers/integration/integration_adapter.go#L179) function that does the same job, so we can use that one instead.

Reply:
Sure. Updated